File: docs-examples/example-java/src/test/java/io/micronaut/nats/docs/jetstream/ProductListener.java
package io.micronaut.nats.docs.jetstream;

import io.micronaut.context.annotation.Requires;
import io.micronaut.nats.jetstream.annotation.JetStreamListener;
import io.micronaut.nats.jetstream.annotation.PushConsumer;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
// end::imports[]

@Requires(property = "spec.name", value = "JetstreamTest")
// tag::clazz[]
@JetStreamListener // <1>
public class ProductListener {

    List<byte[]> messageLengths = Collections.synchronizedList(new ArrayList<>());

    @PushConsumer(value = "myevents", subject = "myevents.>", durable = "test") // <2>
    public void receive(byte[] data) {
        messageLengths.add(data);
    }
}
// end::clazz[]

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/nats/docs/jetstream/ProductListener.kt
package io.micronaut.nats.docs.jetstream


import io.micronaut.context.annotation.Requires

// tag::imports[]
import io.micronaut.nats.jetstream.annotation.JetStreamListener
import io.micronaut.nats.jetstream.annotation.PushConsumer
// end::imports[]
import java.util.*


@Requires(property = "spec.name", value = "JetstreamSpec")
// tag::clazz[]
@JetStreamListener // <1>
class ProductListener {

    val messageLengths: MutableList<ByteArray> = Collections.synchronizedList(ArrayList())

    @PushConsumer(value = "events", subject = "events.>", durable = "test") // <2>
    fun receive(data: ByteArray) {
        messageLengths.add(data)
    }
}
// end::clazz[]
